pkg/generate/app/cmd/newapp.go

```diff
@@ -585,9 +645,73 @@ func makeImageStreamKey(ref kapi.ObjectReference) string {
 	return types.NamespacedName{ref.Namespace, name}.String()
 }
 
+// RunSearch executes the provided config and returns the result of the resolution.
+func (c *AppConfig) RunSearch(out io.Writer) (*SearchResult, error) {
+	c.ensureDockerSearcher()
+	repositories, err := c.individualSourceRepositories()
```

Comment:
It seems you should fail here if any source repos are passed in. I don't see the need to detect code on them either.